src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +22,31 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    private static final int LETTERS_AMOUNT = 5;
+
+    public String getUniqueCharacters(String fileName) throws IOException {
+        TreeSet<Character> lettersFromFile = new TreeSet<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            int input = reader.read();
```

Comment:
consider using readLine instead of read: http://qaru.site/questions/144797/why-is-bufferedreader-read-much-slower-than-readline

Reply:
change type to char.

Reply:
if we'll use while ((input = reader.read()) != -1) then we should leave the int type

Reply:
if we will use readLine() then we need to parse our line too
I will consider that in my next occasions, thak you, i really didn't know that about read() and readLine()